include/chopper/pack/hibf_statistics.hpp

```diff
@@ -0,0 +1,259 @@
+        config{config_},
+        fp_correction{&fp_correction_}
+    {}
+
+    struct bin; // forward decalration
+
+    //!\brief A represetnation of an IBF that gathers information of each bin in an ibf.
+    using ibf = std::vector<bin>;
+
+    //!\brief The kind of bin that is stored.
+    enum class bin_kind
+    {
+        split, //!< A single user bin, split into 1 or more bins (even though 1 is not technically splitted).
+        merged //!< Multiple user bins are merged into a single technical bin.
+    };
+
+    //!\brief Represents a (set) of user bins (see ibf_statistics::bin_kind).
+    struct bin
+    {
+        bin_kind const kind;
+        size_t const cardinality;
+        size_t const num_contained_ubs;
+        size_t const num_spanning_tbs;
+
+        ibf child_ibf;
+
+        bin() = delete; //!< Deleted. Enforce user supplied values for member initialization.
+        bin(bin const & b) = default; //!< Defaulted.
+        bin & operator=(bin const &) = default; //!< Defaulted.
+        bin(bin && b) = default; //!< Defaulted.
+        bin & operator=(bin &&) = default; //!< Defaulted.
+        ~bin() = default; //!< Defaulted.
+
+        bin(bin_kind const kind_,
+            size_t const cardinality_,
+            size_t const num_contained_ubs_,
+            size_t const num_spanning_tbs_) :
+            kind{kind_},
+            cardinality{cardinality_},
+            num_contained_ubs{num_contained_ubs_},
+            num_spanning_tbs{num_spanning_tbs_}
+        {
+            assert((kind == bin_kind::split  && num_contained_ubs == 1ul) ||
+                   (kind == bin_kind::merged && num_spanning_tbs  == 1ul));
+        }
+    };
+
+    //!\brief Prints a tab-indented summary of the statistics of this HIBF to the command line
+    void print_summary()
+    {
+        if (summary.empty())
+            gather_statistics(top_level_ibf, 0);
+
+        std::cout << "\n\tStatistics summary:\n\tlevel\tnum_ibfs\tlevel_size\tlevel_size_no_corr\ttotal_num_tbs"
+                     "\tavg_num_tbs\tsplit_tb_percentage\tmax_split_tb\tavg_split_tb\tmax_factor\tavg_factor\n";
+
+        size_t total_size{};
+        size_t total_size_no_corr{};
+
+        // go through each level and collect and output the statistics
+        for (auto const & [level, s] : summary)
+        {
+            size_t const level_size = std::reduce(s.ibf_mem_size.begin(), s.ibf_mem_size.end());
+            size_t const level_size_no_corr = std::reduce(s.ibf_mem_size_no_corr.begin(), s.ibf_mem_size_no_corr.end());
+
+            total_size += level_size;
+            total_size_no_corr += level_size_no_corr;
+
+            size_t const total_num_tbs = std::reduce(s.num_tbs.begin(), s.num_tbs.end());
+
+            size_t const total_num_split_tbs = std::reduce(s.num_split_tbs.begin(), s.num_split_tbs.end());
+            double const split_tb_percentage = 100 * (double)total_num_split_tbs / (double)total_num_tbs;
```

Comment:
Use `static_cast<double>`; in general it should be enough if either of these values is cast to double.